bin/registerServerTopology.php

```diff
@@ -0,0 +1,283 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once('registerServerTopology-func.php');
+
+/************************ */
+/*     DATA RESOLVING
+/************************ */
+
+/**
+ * Get script params
+ */
+$opt = getopt('u:t:h:n:', ["help::", "dns:", "insecure::"]);
+const SERVER_TYPES = ["poller", "remote", "map", "mbi"];
+
+/**
+ * Format the --help message
+ */
+$helpMessage = <<<'EOD'
+
+Global Options:
+  -u <mandatory>              username of your centreon-web account
+  -t <mandatory>              the server type you want to register:
+            - Poller
+            - Remote
+            - MAP
+            - MBI
+  -h <mandatory>              URL of the Central / Remote Server target platform
+  -n <mandatory>              name of your registered server
+  --help <optional>           get informations about the parameters available
+  --proxy <optional>          provide the differents asked informations
+            - host <mandatory>
+            - port <mandatory>
+            - username <optional>
+            - password <optional>
+  --dns <optional>            provide your DNS instead of automaticaly get server IP
+  --insecure <optional>       allow self-signed certificate
+
+
+EOD;
+
+/**
+ * Display --help message
+ */
+if (isset($opt['help'])) {
+    exit($helpMessage);
+}
+
+/**
+ * Assign options to variables
+ */
+try {
+    if (!isset($opt['u'], $opt['t'], $opt['h'], $opt['n'])) {
+        throw new \InvalidArgumentException(
+            PHP_EOL . 'missing parameter: -u -t -h -n are mandatories:' . PHP_EOL . $helpMessage
+        );
+    }
+
+    $username = $opt['u'];
+    $serverType = in_array(strtolower($opt['t']), SERVER_TYPES)
+        ? strtolower($opt['t'])
+        : false;
+
+    if (!$serverType) {
+        throw new \InvalidArgumentException(
+            "-t must be one of those value"
+            . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
+        );
+    }
+
+    if (isset($opt['dns'])) {
+        $dns = filter_var($opt['dns'], FILTER_VALIDATE_DOMAIN);
+        if (!$dns) {
+            throw new \InvalidArgumentException(
+                PHP_EOL . "Bad DNS Format" . PHP_EOL
+            );
+        }
+    }
+
+    $targetHost = $opt['h'];
+    $serverHostName = $opt['n'];
+} catch (\InvalidArgumentException $e) {
+    exit($e->getMessage());
+}
+$password = askQuestion($targetHost . ': enter your password ', true);
+$proxy =  strtolower(askQuestion("Are you using a proxy ? (y/n)"));
+/**
+ * Parsing url part from params -h
+ */
+$targetURL = parse_url($targetHost);
+$protocol = $targetURL['scheme'] ?? 'http';
```

Comment:
if only http or https are allowed (eg : no ftp / ftps or other)
```suggestion
$protocol = ('https' === $targetURL['scheme'] ? 'https' : 'http');
```

Reply:
this will throw an error if not scheme are send, e.g if i only send an IP i will get:
Undefined index: scheme in /usr/share/centreon/bin/registerServerTopology.php on line 109.

Here we assign the scheme (http/https/ftp or whatever) if he is present in the URL, if not by default we set http